test/unit/modules/blocks/utils.js

```diff
@@ -13,268 +13,558 @@
+				);
+				expect(err).to.equal('Blocks#getIdSequence error');
+				expect(sequence).to.be.undefined;
+				done();
 			});
+		});
 
-			it(
-				'should add the genesis block to the end of the block array, if it does not contain it already'
-			);
-
-			it(
-				'should add the last block to the beginning of the block array, if it does not contain it already'
-			);
-
-			it('should call callback with error = null');
-
-			describe('result object', () => {
-				it('should assign firstHeight to the height of the last block');
+		it('should return error when no row is found', done => {
+			library.db.blocks.getIdSequence = sinonSandbox.stub().resolves([]);
 
-				it('should assign ids to a string of the block ids');
+			blocksUtilsModule.getIdSequence(10, (err, sequence) => {
+				expect(err).to.equal('Failed to get id sequence for height: 10');
+				expect(sequence).to.be.undefined;
+				done();
 			});
+		});
 
-			it('should call callback with result object');
+		it('should return valid block id list', done => {
+			library.db.blocks.getIdSequence = sinonSandbox
+				.stub()
+				.resolves([
+					{ id: 1, height: 2 },
+					{ id: 2, height: 3 },
+					{ id: 3, height: 4 },
+					{ id: 4, height: 5 },
+				]);
+
+			blocksUtilsModule.getIdSequence(10, (err, sequence) => {
+				expect(sequence).to.be.an('object');
+				expect(sequence.firstHeight).to.equal(1);
+				expect(sequence.ids).to.equal(
+					'9314232245035524467,1,2,3,4,6524861224470851795'
+				);
+				done();
+			});
 		});
 	});
 
 	describe('loadBlocksData', () => {
-		describe('when 3 parameters are passed to loadBlocksData', () => {
-			it('should call third parameter');
+		it('should return error when library.db.blocks.loadBlocksData fails', done => {
+			library.db.blocks.getHeightByLastId = sinonSandbox.stub().resolves(null);
+
+			blocksUtilsModule.loadBlocksData({ id: '1' }, (err, blocks) => {
+				expect(loggerStub.error.args[0][0]).to.contains(
+					"TypeError: Cannot read property 'length' of null"
+				);
+				expect(err).to.equal('Blocks#loadBlockData error');
+				expect(blocks).to.be.undefined;
+				done();
+			});
 		});
 
-		describe('when 2 parameters are passed to loadBlocksData', () => {
-			it('should call second parameter');
-		});
+		it('should return error when called with both id and lastId', done => {
+			library.db.blocks.getHeightByLastId = sinonSandbox.stub().resolves(['1']);
 
-		describe('when filter.id and filter.lastId are defined', () => {
-			it('should call callback with Invalid Filter error');
+			blocksUtilsModule.loadBlocksData(
+				{ id: '1', lastId: '5' },
+				(err, blocks) => {
+					expect(err).to.equal('Invalid filter: Received both id and lastId');
+					expect(blocks).to.be.undefined;
+					done();
+				}
+			);
 		});
 
-		describe('when filter.lastId is undefined and filter.id is defined', () => {
-			it('should set params.id to filter.id');
+		it('should return empty row when called with invalid id', done => {
```

Comment:
Where there is a check for that in tested function?

Reply:
This is the behavior when filtering using an unknown id.

Reply:
Ok, got it, that behavior is defined in `beforeEach` stubs.